traits/util/tests/test_trait_documenter.py

```diff
@@ -26,25 +30,46 @@ def _python_version_is_32():
 class TestTraitDocumenter(unittest.TestCase):
     """ Tests for the trait documenter. """
 
-    def setUp(self):
-        self.source = """
-    depth_interval = Property(Tuple(Float, Float),
-                              depends_on="_depth_interval")
-"""
-        string_io = StringIO.StringIO(self.source)
-        tokens = tokenize.generate_tokens(string_io.readline)
-        self.tokens = tokens
-
     def test_get_definition_tokens(self):
 
         from traits.util.trait_documenter import _get_definition_tokens
 
-        definition_tokens = _get_definition_tokens(self.tokens)
+        src = textwrap.dedent("""\
+        depth_interval = Property(Tuple(Float, Float),
+                                  depends_on="_depth_interval")
+        """)
+        string_io = StringIO.StringIO(src)
+        tokens = tokenize.generate_tokens(string_io.readline)
+
+        definition_tokens = _get_definition_tokens(tokens)
 
         # Check if they are correctly untokenized. This should not raise.
         string = tokenize.untokenize(definition_tokens)
 
-        self.assertEqual(self.source.rstrip(), string)
+        self.assertEqual(src.rstrip(), string)
+
+    def test_add_line(self):
+
+        from traits.util.trait_documenter import TraitDocumenter
+
+        src = textwrap.dedent("""\
+        class Fake(HasTraits):
+
+            #: Test attribute
+            test = Property(Bool, label=u"ミスあり")
+        """)
+        mocked_directive = mock.MagicMock()
+
+        documenter = TraitDocumenter(mocked_directive, 'test', '   ')
+        documenter.object_name = 'Property'
+
+        with mock.patch(
+                'traits.util.trait_documenter.inspect.getsource',
+                return_value=src):
+            with mock.patch(
+                    'traits.util.trait_documenter.ClassLevelDocumenter'):
+                documenter.add_directive_header('')
```

Comment:
Is there any positive side-effect or result that we can test here? It looks as though at the moment this is simply testing that nothing goes wrong.

Reply:
Thanks for the review @mdickinson 
Yup, this is just testing that a `UnicodeDecodeError` is not raised. I couldn't think of a way to test a positive result without implementing an entire `directive` for the `ClassLevelDocumenter`, which AFAICT would take a lot of effort. I updated the test to only mock a specific method of `ClassLevelDocumenter`, and asserted that `directive.result.append` was called.

From `sphinx.ext.autodoc.Documenter`:

```
    def add_line(self, line, source, *lineno):
        # type: (unicode, unicode, int) -> None
        """Append one line of generated reST to the output."""
        self.directive.result.append(self.indent + line, source, *lineno)
```

Any suggestions on how we can test this further?